app/models/account.rb

```diff
@@ -118,6 +118,7 @@ class Account < ApplicationRecord
   validates :display_name, length: { maximum: DISPLAY_NAME_LENGTH_LIMIT }, if: -> { local? && will_save_change_to_display_name? }
   validates :note, note_length: { maximum: NOTE_LENGTH_LIMIT }, if: -> { local? && will_save_change_to_note? }
   validates :fields, length: { maximum: DEFAULT_FIELDS_SIZE }, if: -> { local? && will_save_change_to_fields? }
+  validates_with EmptyProfileFieldNamesValidator, if: -> { local? && will_save_change_to_fields? }
```

Comment:
Should add at least 1-2 new examples to model spec showing the changing validation behaviour here, and optionally a request spec showing the changing response json (and http code, if it does change) around the scenario we are modifying here.

Reply:
Thanks for pointing that out, completely forgot about the specs. I have added examples in the account model spec for too many fields, fields with values but no names, and the valid name but no value scenario. As for the request I don't believe that behavior has changed, if you submit a no-name field you still get back a 200 because the error is just rendered in the page.